wordpress-mobile/WordPress-Android
# Removes sections for post search results

This PR removes sections for post search results by reverting changes in #10837. Just to be on the safe side I did the revert manually since we made some changes to this logic afterwards.

The only thing that I didn't revert yet is the ability to search for all post statuses. Previous to the sectioning PR, we were only searching for `PUBLISHED` and `PRIVATE` posts as it can be seen [here](https://github.com/wordpress-mobile/WordPress-Android/pull/10837/files#diff-89f2d36ef519c45b6bfa8348e2982681L12). I didn't know if we wanted to change that back, but if we want to keep it as is, there is one design concern. We don't allow opening the editor for trashed posts until they are moved to drafts. That's very obvious when we do it in the `Trashed` post list, but not so much with the compact design of post search results. For example, in the screenshot below, the last result is for `TRASHED` post. 

<img src=https://user-images.githubusercontent.com/662023/74362805-59c98180-4d97-11ea-9562-b137c64c4d46.png width=300>

@SylvesterWilmott @osullivanchris @malinajirka I know this was mentioned in the internal discussion, but I don't think we landed a solution for this particular issue. Could you all let me know your thoughts on this? Thanks!

**To test:**

* Go to post list
* Search for some query
* Verify the results are what you'd expect to see.
* Test searching for posts in different statuses.
* Test searching for posts that are not yet downloaded in the post list (for example in a fresh installation search for scheduled posts, or posts that are older etc)

**PR submission checklist:**
